src/components/sources/OlSourceWebglVector.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+  <div>
+    <slot></slot>
+  </div>
+</template>
+
+<script setup lang="ts">
+import VectorSource, { type Options } from "ol/source/Vector";
+import type { Ref } from "vue";
+import { inject, watch, onMounted, onUnmounted, provide, computed } from "vue";
+
+import Layer from "ol/layer/Layer.js";
+import WebGLVectorLayerRenderer from "ol/renderer/webgl/VectorLayer.js";
+class WebGLVectorLayer extends Layer {
+  createRenderer() {
+    return new WebGLVectorLayerRenderer(this, {
+      className: this.getClassName(),
+      style: properties.style,
+    });
+  }
+}
+
+import type Line from "ol/geom/Line";
+import type Polygon from "ol/geom/Polygon";
+
+import usePropsAsObjectProperties from "@/composables/usePropsAsObjectProperties";
+import {
+  FEATURE_EVENTS,
+  useOpenLayersEvents,
+} from "@/composables/useOpenLayersEvents";
+
+// prevent warnings caused by event pass-through via useOpenLayersEvents composable
+defineOptions({
+  inheritAttrs: false,
+});
+
+const props = withDefaults(defineProps<Options<Line|Polygon>>(), {   // SHOULD prob. not be Line
+  overlaps: true,
+  projection: "EPSG:3857",
+  useSpatialIndex: true,
+  wrapX: true,
+});
+
+const layer = inject<Ref<WebGLVectorLayer<VectorSource<Line|Polygon>>> | null>(   // SHOULD prob. not be Line
```

Comment:
It should / could be [`SimpleGeometry`](https://openlayers.org/en/latest/apidoc/module-ol_geom_SimpleGeometry-SimpleGeometry.html) to be more flexible right?

Reply:
Does that mean that in `sources/OlSourceWebglVector.vue` that if it had:
```
import WebGLVectorLayerRenderer from "ol/renderer/webgl/VectorLayer";

OlSourceWebglVector.vueclass WebGLVectorLayer extends Layer {
  createRenderer() {
    return new WebGLVectorLayerRenderer(this, {
      className: this.getClassName(),
      style: properties.style,
    });
  }
}
```
and:
```
const layer = inject<Ref<WebGLVectorLayer<VectorSource<SimpleGeometry>>> | null>
```
that it should work for all geom types?   I'll try doing that.
(That would mean that all of the current WebglPoints items would be taken over by this WebglVector one, right?)

Reply:
I refactored it a little bit and extracted the class. Should work now.